### ISSUE TITLE
`approx_const` misses constants when precise

### ISSUE THREAD

**Issue by teofr** [rust-lang/rust-clippy#15194]
### Summary

`approx_const` is missing some constants when they are either too precise (since truncation messes things up) or when the value has leading 0s.

I believe both these things can be fixed by comparing the float representation of the literal rather than the symbol.

[Playground code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&code=%0Afn+main%28%29+%7B%0A++++%2F%2F+pi+declared+3.14159265358979323846264338327950288_f64%0A++++%0A++++%23%5Ballow%28clippy%3A%3Aexcessive_precision%29%5D%0A++++let+x%3A+f64+%3D+3.14159265358979323846264338327950288%3B%0A++++dbg%21%28x%29%3B%0A++++%0A++++%23%5Ballow%28clippy%3A%3Aexcessive_precision%29%5D%0A++++let+y%3A+f64+%3D+003.14159265358979311599796346854418516159057617187500%3B%0A++++%2F%2F+The+one+below+does+trigger%0A++++%2F%2F+let+y%3A+f64+%3D+3.14159265358979311599796346854418516159057617187500%3B%0A++++dbg%21%28y%29%3B%0A%7D%0A)

### Lint Name

approx_const

### Reproducer

I tried this code:

```rust
fn main() {
    // pi declared 3.14159265358979323846264338327950288_f64 in std.core
    
    #[allow(clippy::excessive_precision)]
    let x: f64 = 3.14159265358979323846264338327950288;
    dbg!(x);
    
    #[allow(clippy::excessive_precision)]
    let y: f64 = 003.14159265358979311599796346854418516159057617187500;
    // The one below does trigger
    // let y: f64 = 3.14159265358979311599796346854418516159057617187500;
    dbg!(y);
}
```

I expected to see this happen:

`approx_const` lint on both `x` and `y` definitions

Instead, this happened:

No lints

### Version

```text
rustc 1.90.0-nightly (0fa4ec6cd 2025-06-25)
binary: rustc
commit-hash: 0fa4ec6cde46fa17ab07acb5531cfe0dc1349ffa
commit-date: 2025-06-25
host: aarch64-apple-darwin
release: 1.90.0-nightly
LLVM version: 20.1.7
```
